### DISCUSSION TITLE
How to support per unit bucket boundaries in v0.33.0?

### DISCUSSION THREAD

**Discussion by yux0** [open-telemetry/opentelemetry-go#3412] [General]
Currently, we are on Go sdk v0.30.0 and implemented a customized aggregator selector to support per unit bucket boundaries. [sample](https://github.com/temporalio/temporal/blob/master/common/metrics/opentelemetry_aggregator_selector.go#L58).

I am trying to upgrade to v0.33.0 and do not find a way to matchInstrumentUnit. Does it support of per unit bucket boundaries in v0.33.0?

**Comment by MrAlias** [comment]
In the new SDK this would be accomplished with a [`view.View`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric/view), in theory. It is currently theoretical because that package does not support matching instrument units.

There is currently [a proposal](https://github.com/open-telemetry/opentelemetry-go/issues/3399) to refactor that views. The proposed replacement would allow you to accomplish matching an instrument unit and replacing histogram boundaries. Please leave a :+1: on the proposal if you would like to support its inclusion or any comment you have.

**Comment by yiminc** [comment]
@MrAlias , how to do per unit type custom histogram boundaries? The proposed change related to this requirement is to `replace the view.View with a View func`. The View func is defined as `type View func(Instrument) (Stream, bool)`. However, the Instrument does not have unit info? So how to match based on unit?

**Reply by MrAlias** [reply]
The `Instrument` type would be defined to have a unit field: https://github.com/open-telemetry/opentelemetry-go/pull/3397/files#diff-2b58257199790c9227975e80f5f2122187de28546546271137da97e324be4782R80

**Reply by MrAlias** [reply]
I've updated the proposal issue to reflect this.
